src/components/CityListItem.js

```diff
@@ -1,59 +1,92 @@
 import React, { Component } from 'react';
-import {
-  View,
-  Text,
-  TouchableOpacity,
-  StyleSheet,
-  ActivityIndicator,
-  Image,
-} from 'react-native';
-import * as WeatherDataUtil from '../utils/WeatherDataUtil';
+import { View, Text, TouchableOpacity, StyleSheet, ActivityIndicator, Image } from 'react-native';
+import PropTypes from 'prop-types';
+import * as weatherData from '../service/weatherData';
 
-const CityListItem = props => {
-  const weather = props.weatherReport.weather[0];
+class CityListItem extends Component {
+  static propTypes = {
+    city: PropTypes.object,
+  };
 
-  return (
-    <View style={style.citySection}>
-      <TouchableOpacity onPress={props.onPress}>
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoading: false,
+      weatherReport: {},
+    };
+  }
+
+  async componentDidMount() {
+    const { city } = this.props;
+    try {
+      this.setState({
+        isLoading: true,
+      });
+
+      const weatherReport = await weatherData.fetchWeatherData(city);
+
+      this.setState({
+        weatherReport,
+      });
+    } finally {
+      this.setState({
+        isLoading: false,
+      });
+    }
+  }
+
+  renderInformation() {
+    const { isLoading, weatherReport } = this.state;
+
+    if (isLoading) {
+      return (
         <View>
-          <Text style={style.cityName}>{props.city.value}</Text>
+          <ActivityIndicator size="large" color="#000000" />
         </View>
-      </TouchableOpacity>
+      );
+    }
 
-      {!props.isLoading ? (
-        <View style={style.cityInfo}>
-          <Image
-            style={{ width: 50, height: 50 }}
-            source={{
-              uri: 'https://openweathermap.org/img/w/' + weather.icon + '.png',
-            }}
-          />
+    return (
+      <View style={style.cityInfo}>
+        <Image
+          style={style.weatherIcon}
+          source={{
+            uri: `https://openweathermap.org/img/w/${weatherReport.icon}.png`,
+          }}
+        />
 
-          <Text style={style.infoText}>{weather.main}</Text>
+        <Text style={style.infoText}>
+          {weatherReport.weather_description}
+        </Text>
 
-          <View>
-            <Text style={style.infoText}>
-              min:{' '}
-              {WeatherDataUtil.convertKelvinToCelsius(
-                props.weatherReport.main.temp_min
-              )}
-            </Text>
-            <Text style={style.infoText}>
-              max:{' '}
-              {WeatherDataUtil.convertKelvinToCelsius(
-                props.weatherReport.main.temp_max
-              )}
-            </Text>
-          </View>
-        </View>
-      ) : (
         <View>
-          <ActivityIndicator size="large" color="#000000" />
+          <Text style={style.infoText}>
+            {`min: ${weatherReport.temp_min}`}
```

Comment:
extract weather props at the top of the function like this: 
`const { weatherReport: { tempMin, tempMax, icon, description } } = this.state;`